pygmt/helpers/utils.py

```diff
@@ -192,7 +192,58 @@ def data_kind(
     return kind
 
 
-def non_ascii_to_octal(argstr: str) -> str:
+def check_encoding(argstr: str) -> str:
```

Comment:
Would it be too much to have a typehint like this?

```suggestion
def check_encoding(argstr: str) -> Literal[
    "ascii",
    "ISOLatin1+",
    "ISO-8859-1",
    "ISO-8859-2",
    "ISO-8859-3",
    "ISO-8859-4",
    "ISO-8859-5",
    "ISO-8859-6",
    "ISO-8859-7",
    "ISO-8859-8",
    "ISO-8859-9",
    "ISO-8859-10",
    "ISO-8859-11",
    "ISO-8859-13",
    "ISO-8859-14",
    "ISO-8859-15",
    "ISO-8859-16",
    "ISO-8859-17",
]:
```

Reply:
I'm thinking about that too. Actually the same types can be used in typing hints the `encoding` parameter of the `non_ascii_to_octal` function. So my initial plan is to define a generic type
```
type Encodings = Literal[
    "ascii",
    "ISOLatin1+",
    "ISO-8859-1",
    "ISO-8859-2",
    "ISO-8859-3",
    "ISO-8859-4",
    "ISO-8859-5",
    "ISO-8859-6",
    "ISO-8859-7",
    "ISO-8859-8",
    "ISO-8859-9",
    "ISO-8859-10",
    "ISO-8859-11",
    "ISO-8859-13",
    "ISO-8859-14",
    "ISO-8859-15",
    "ISO-8859-16",
]
```
and then use it like:
```
def check_encoding(argstr: str) -> Encodings:
def non_ascii_to_octal(argstr: str, encoding: Encodings = "ISOLatin1+") -> str:
```
but mypy v1.10.0 complains
```
pygmt/helpers/utils.py:21: error: PEP 695 type aliases are not yet supported  [valid-type]
```
mypy v1.11.0 started to support PEP 695 but it was released just a few days ago (https://mypy-lang.blogspot.com/2024/07/mypy-111-released.html) and the PEP 695 support is still experimental.

Reply:
Done in 7607c7efb0b2442b1647239d51e475ebf62284e2.